clients/da-vinci-client/src/main/java/com/linkedin/davinci/config/VeniceServerConfig.java

```diff
@@ -632,6 +646,23 @@ public VeniceServerConfig(VeniceProperties serverProperties, Map<String, Map<Str
     ingestionMemoryLimit = extractIngestionMemoryLimit(serverProperties, ingestionMode, forkedProcessJvmArgList);
     LOGGER.info("Ingestion memory limit: {} after subtracting other usages", ingestionMemoryLimit);
     ingestionMlockEnabled = serverProperties.getBoolean(INGESTION_MLOCK_ENABLED, false);
+
+    try {
+      String producerFactoryClassName = serverProperties
```

Comment:
Any particular reason to pass client factories as configs instead of actual instance of `PubSubClientsFactory` dependency to `venice-server` and `venice-controller`?

Reply:
The reason is: for `DavinciBackend` and `IngestionIsolationServer`, the configuration information is most from `VeniceServerConfig`. As for `PubSubClientsFactory`, it is not meaningful class name we can put into the configs as string, so I need this concrete facotroies name. This reminds me one thing, for integration test, we also need to add this into configs to be passed to `IngestionIsolationServer` or `DavinciBackend`.

Reply:
For II and DVC we can pass them as configs but for controller and server, I don't think we need to pass them as configs.

Reply:
> For II and DVC we can pass them as configs but for controller and server, I don't think we need to pass them as configs.

This will mean we'll have two diff ways to pass pus-sub adapters which may not be ideal. Maybe it's better to just use a config-based approach.

Reply:
Yeah, this change supports a config-based approach. Inside `VeniceControllerWrapper`  and `VeniceServerWrapper` class, there are some logics to get adapter factory class from `PubSubClientsFactory` and put them into configs. The benefits will be: 1) we will not need to add logic inside our internal project to prepare `PubSubClientsFactory` , 2)For II and DVC, they can leverage these configs naturally.